Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -79,11 +79,38 @@ public void DeserializesObjectsWithIgnoreDefaults()
             var serializer = new IonSerializer(new IonSerializationOptions {IgnoreDefaults = true});
             var deserialized = serializer.Deserialize<Motorcycle>(stream);
             
-            Assert.IsNull(deserialized.Brand);
+            Assert.IsNull(deserialized.Make);
             Assert.IsNull(deserialized.color);
             Assert.IsNotNull(deserialized.canOffroad);
         }
 
+        [TestMethod]
+        public void SerializesAndDeserializesObjectsWithCaseInsensitiveProperties()
+        {
+            var serializer = new IonSerializer(new IonSerializationOptions { PropertyNameCaseInsensitive = true });
+            var stream = serializer.Serialize(TestObjects.Harley);
+
+            var deserialized1 = serializer.Deserialize<MOTORCYCLE>(stream);
+            
+            Assert.AreEqual(TestObjects.Harley.Make, deserialized1.MAKE);
+            Assert.IsNull(deserialized1.COLOR);
+            Assert.IsFalse(deserialized1.CANOFFROAD);
+
+            stream.Position = 0;
+            var deserialized2 = serializer.Deserialize<motorcycle>(stream);
+            
+            Assert.AreEqual(TestObjects.Harley.Make, deserialized2.make);
+            Assert.IsNotNull(deserialized2.color);
+            Assert.IsFalse(deserialized2.canoffroad);
+            
+            stream.Position = 0;
+            var deserialized3 = serializer.Deserialize<MoToRcYcLe>(stream);
```

Comment:
This is not right. We don't care about type names, we care about property names.

Reply:
This class is called `MoToRcYcLe` not because we're testing the type name, but because this class has property/field names with that case style ie. (`MaKe`, `CoLoR`, `CaNoFfRoAd`). I could have named it anything but ultimately went with `MoToRcYcLe` to indicate that it's the same as the `Motorcycle` class but with that case style.
Likewise with the classes `MOTORCYLCE` and `motorcycle`.